src/sa1hybridizer/index.js

```diff
@@ -0,0 +1,396 @@
+    !map16_hi_by = $7F0000
+    !save_mem = $700000
+endif"
+`;
+
+const sprite_addr_list = [0x7fab10, 0x7fab1c, 0x7fab28, 0x7fab34, 0x7fab9e, 0x7fab40, 0x7fab4c, 0x7fab58, 0x7fab64, 0x7fac00,
+    0x7fac08, 0x7fac10, 0x9e, 0xaa, 0xb6, 0xc2, 0xd8, 0xe4, 0x14c8, 0x14d4, 0x14e0, 0x14ec, 0x14f8,
+    0x1504, 0x1510, 0x151c, 0x1528, 0x1534, 0x1540, 0x154c, 0x1558, 0x1564, 0x1570, 0x157c, 0x1588,
+    0x1594, 0x15a0, 0x15ac, 0x15b8, 0x15c4, 0x15d0, 0x15dc, 0x15ea, 0x15f6, 0x1602, 0x160e, 0x161a,
+    0x1626, 0x1632, 0x163e, 0x164a, 0x1656, 0x1662, 0x166e, 0x167a, 0x1686, 0x186c, 0x187b, 0x190f,
+    0x1938, 0x7faf00, 0x1fd6, 0x1fe2];
+
+const WordType = {
+    OTHER: -1,
+    ADDR: 1,
+    COMMA: 2
+}
+
+class WordTuple {
+    constructor(type, word, index) {
+        this.type = type;
+        this.word = word;
+        this.index = index;
+    }
+}
+
+Object.freeze(WordType);
+
+function normalizeHexStringToHexString(str) {
+    // this seems rather useless (and it may be) but it ensures consitency
+    return parseInt(str, 16).toString(16).toUpperCase();
+}
+
+function check_bwram(word) {
+    let bwram_word = parseInt(word, 16)
+    let bwram_remapped_list = [0x7F9A7B, 0x7027FF]          // Wiggler's segment buffer, Expansion area planned for SMW hacks
+    let map16_lo_by = [0x7EC800, 0x7EFFFF]                // Map16 low byte plus Overworld related data.
+    let map16_hi_by = [0x7FC800, 0x7FFFFF]                // Map16 high byte.
+    let save_mem = [0x700000, 0x7007FF]                // Original save memory (2 kB big). Not everything is used
+    let bwram_list = [map16_lo_by, map16_hi_by, save_mem]
+    let bwram_indexes = bwram_list.map(x => {
+        let [b, e] = x;
+        return b <= bwram_word <= e;
+    });
+    let subs = ['map16_lo_by', 'map16_hi_by', 'save_mem']
+    if (bwram_indexes.some(x => x)) {
+        let true_index = bwram_indexes.indexOf(True)
+        let sub = `${bwram_word.toString(16).toUpperCase().padStart(6, '0')}&$00FFFF|!${subs[true_index]}`
+        return [sub, true]
+    }
+    else if (bwram_word in bwram_remapped_list) {
+        let sub = `!${bwram_word.toString(16).toUpperCase().padStart(6, '0')}`
+        return [sub, true]
+    }
+    return [word, false]
+}
+
+function process_word(word, index, splitted, comma_index, messages) {
+    let requires_manual_conversion = false;
+    let converted = true;
+    let add_dp = false;
+    if (comma_index !== -1) {
+        if (word.length == 4 && (splitted[comma_index + 1] === 'y' || splitted[comma_index + 1] === 'x') && word.substring(0, 2) == '00')
+            add_dp = true;
+    }
+    if (word.startsWith('8') && word.length == 6) {
+        word = word.replace('8', '0', 1);
+    }
+    let numeric_word = undefined;
+    try {
+        numeric_word = parseInt(word, 16);
+    } catch (err) {
+        throw err;
+    }
+    let bwram_define_needed = false;
+    if (word.length === 6)
+        word, bwram_define_needed = check_bwram(word);
```

Comment:
JS interprets this as two separate expressions, `word` (which is essentially a no-op) and `bwram_define_needed = check_bwram(word);`. Since `check_bwram` returns an array, `bwram_define_needed` will always be truthy. I assume you meant to use a destructuring syntax instead.
```suggestion
        [word, bwram_define_needed] = check_bwram(word);
```

Reply:
Fixed in 010b017cb417f4e3d9f01b938abf4a2a9697f87c